projects/packages/sync/src/modules/class-woocommerce-product-meta-lookup.php

```diff
@@ -0,0 +1,356 @@
+			return $args;
+		}
+
+		$product_id = $args[0];
+
+		// Get the product lookup data
+		$lookup_data = $this->get_product_lookup_by_ids( array( $product_id ) );
+
+		if ( ! empty( $lookup_data ) ) {
+			$args[1] = reset( $lookup_data ); // Get the first (and only) result
+		}
+
+		return $args;
+	}
+
+	/**
+	 * Enqueue the WooCommerce Product Lookup actions for full sync.
+	 *
+	 * @access public
+	 *
+	 * @param array   $config               Full sync configuration for this sync module.
+	 * @param int     $max_items_to_enqueue Maximum number of items to enqueue.
+	 * @param boolean $state                True if full sync has finished enqueueing this module, false otherwise.
+	 * @return array Number of actions enqueued, and next module state.
+	 */
+	public function enqueue_full_sync_actions( $config, $max_items_to_enqueue, $state ) {
+		return $this->enqueue_all_ids_as_action(
+			'jetpack_full_sync_woocommerce_product_meta_lookup',
+			$this->table(),
+			'product_id',
+			$this->get_where_sql( $config ),
+			$max_items_to_enqueue,
+			$state
+		);
+	}
+
+	/**
+	 * Retrieve an estimated number of actions that will be enqueued.
+	 *
+	 * @access public
+	 *
+	 * @param array $config Full sync configuration for this sync module.
+	 * @return int Number of items yet to be enqueued.
+	 */
+	public function estimate_full_sync_actions( $config ) {
+		global $wpdb;
+
+		$query = "SELECT count(*) FROM {$this->table()} WHERE " . $this->get_where_sql( $config );
+		// phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared,WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching
+		$count = (int) $wpdb->get_var( $query );
+
+		return (int) ceil( $count / self::ARRAY_CHUNK_SIZE );
+	}
+
+	/**
+	 * Return a list of objects by their type and IDs
+	 *
+	 * @param string $object_type Object type.
+	 * @param array  $ids IDs of objects to return.
+	 *
+	 * @access public
+	 *
+	 * @return array|object|WP_Error|null
+	 */
+	public function get_objects_by_id( $object_type, $ids ) {
+		if ( 'product_meta_lookup' !== $object_type || empty( $ids ) || ! is_array( $ids ) ) {
+			return array();
+		}
+
+		return $this->get_product_lookup_by_ids( $ids );
+	}
+
+	/**
+	 * Returns a list of product lookup objects by their IDs.
+	 *
+	 * @param array  $ids List of product IDs to fetch.
+	 * @param string $order Either 'ASC' or 'DESC'.
+	 *
+	 * @access public
+	 *
+	 * @return array|object|null
+	 */
+	public function get_product_lookup_by_ids( $ids, $order = '' ) {
+		global $wpdb;
+
+		if ( ! is_array( $ids ) ) {
+			return array();
+		}
+
+		// Make sure the IDs are numeric and are non-zero.
+		$ids = array_filter( array_map( 'intval', $ids ) );
+
+		if ( empty( $ids ) ) {
+			return array();
+		}
+
+		// Prepare the placeholders for the prepared query below.
+		$placeholders = implode( ',', array_fill( 0, count( $ids ), '%d' ) );
+
+		$query = "SELECT * FROM {$this->table()} WHERE product_id IN ( $placeholders )";
```

Comment:
Unlike previous PR in WooCommerce Analytics, only sync data that's needed for analytics. We're syncing all data here.